### PR TITLE
Fix management of seccomp ConfigMap when not used

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

Currently the operator always mount the seccomp volumes, even if we use a seccomp profile from host (e.g. no install performed in init container), while the ConfigMap is not created, creating PODs referencing non-existing ConfigMaps.
Also, the ConfigMap was never deleted if setting was changed from using seccomp profile from ConfigMap to using profile from host.

Took the opportunity to change the way we do diffs in tests.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Deploy the Agent with `system-probe` enabled. Then change the DatadogAgent set the `SecCompProfileName` attribute to `unconfined`. The new DaemonSet should start properly and the ConfigMap named `<dda.Name>-system-probe-seccomp` should be deleted.

### CODE REVIEW

**Review by clamoriniere — APPROVED** [review]

File: config/crd/bases/v1/datadoghq.com_datadogagents.yaml [hunk 1]
```diff
@@ -3630,7 +3630,8 @@ spec:
                         type: object
                       secCompCustomProfileConfigMap:
                         description: SecCompCustomProfileConfigMap specify a pre-existing
-                          ConfigMap containing a custom SecComp profile.
+                          ConfigMap containing a custom SecComp profile. This ConfigMap
+                          must contain a file named system-probe-seccomp.json
```
Comment (jtappa) [inline config/crd/bases/v1/datadoghq.com_datadogagents.yaml:3634]:
```suggestion
                          must contain a file named system-probe-seccomp.json.
```

File: config/crd/bases/v1beta1/datadoghq.com_datadogagents.yaml [hunk 2]
```diff
@@ -3492,7 +3492,8 @@ spec:
                       type: object
                     secCompCustomProfileConfigMap:
                       description: SecCompCustomProfileConfigMap specify a pre-existing
-                        ConfigMap containing a custom SecComp profile.
+                        ConfigMap containing a custom SecComp profile. This ConfigMap
+                        must contain a file named system-probe-seccomp.json
```
Comment (jtappa) [inline config/crd/bases/v1beta1/datadoghq.com_datadogagents.yaml:3496]:
```suggestion
                        must contain a file named system-probe-seccomp.json.
```

File: docs/configuration.md [hunk 3]
```diff
@@ -175,7 +175,7 @@ spec:
 | agent.systemProbe.env | The Datadog SystemProbe supports many environment variables. See also: https://docs.datadoghq.com/agent/docker/?tab=standard#environment-variables |
 | agent.systemProbe.resources.limits | Limits describes the maximum amount of compute resources allowed. More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/ |
 | agent.systemProbe.resources.requests | Requests describes the minimum amount of compute resources required. If Requests is omitted for a container, it defaults to Limits if that is explicitly specified, otherwise to an implementation-defined value. More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/ |
-| agent.systemProbe.secCompCustomProfileConfigMap | SecCompCustomProfileConfigMap specify a pre-existing ConfigMap containing a custom SecComp profile. |
+| agent.systemProbe.secCompCustomProfileConfigMap | SecCompCustomProfileConfigMap specify a pre-existing ConfigMap containing a custom SecComp profile. This ConfigMap must contain a file named system-probe-seccomp.json |
```
Comment (jtappa) [inline docs/configuration.md:178]: I'm assuming this is autogenerated and would be fixed with above suggestion?

**Review by jtappa — CHANGES_REQUESTED** [review]

File: docs/configuration.md (hunk 3 above)
Comment (vboulineau) [reply docs/configuration.md]: That's correct!

**Pull request merged**